File: src/MaksIT.CertsUI/Models/APIKeys/Search/SearchApiKeyEntityScopeResponse.cs
using MaksIT.Core.Abstractions.Webapi;
using MaksIT.CertsUI.Engine;

namespace MaksIT.CertsUI.Models.APIKeys.Search;

/// <summary>
/// One API key entity scope row in search results (includes parent API key description).
/// </summary>
public class SearchApiKeyEntityScopeResponse : ResponseModelBase {
  public required Guid Id { get; set; }
  public required Guid ApiKeyId { get; set; }
  public string? Description { get; set; }
  public required Guid EntityId { get; set; }
  public string? EntityName { get; set; }
  public ScopeEntityType EntityType { get; set; }
  public ScopePermission Scope { get; set; }
}

File: src/MaksIT.WebUI/src/models/certsUI/apiKeys/search/SearchApiKeyEntityScopeResponse.ts
import { ResponseModelBase } from '../../../ResponseModelBase'
import { ScopeEntityType, ScopePermission } from '../../../engine/scopeEnums'

/** Mirrors <c>MaksIT.CertsUI.Models.APIKeys.Search.SearchApiKeyEntityScopeResponse</c>. */
export interface SearchApiKeyEntityScopeResponse extends ResponseModelBase {
  id: string
  apiKeyId: string
  description?: string | null
  entityId: string
  entityName?: string | null
  entityType: ScopeEntityType
  scope: ScopePermission
}
